src/main/java/net/imglib2/algorithm/hough/HoughTransforms.java

```diff
@@ -0,0 +1,446 @@
+			final RandomAccessibleInterval< T > input,
+			final RandomAccessibleInterval< U > votespace,
+			final int nTheta,
+			final int nRho,
+			final T threshold )
+
+	{
+		final Predicate< T > p = o -> threshold.compareTo( o ) < 0;
+		voteLines( input, votespace, nTheta, nRho, p );
+	}
+
+	/**
+	 *
+	 * Runs a Hough Line Tranform on an image and populates the vote space
+	 * parameter with the results.
+	 * <p>
+	 * Vote space here has two dimensions: {@code rho} and {@code theta}.
+	 * {@code theta} is measured in radians {@code [-pi/2 pi/2)}, {@code rho} is
+	 * measured in {@code [-rhoMax, rhoMax)}.
+	 * </p>
+	 * <p>
+	 * Lines are modeled as
+	 * </p>
+	 *
+	 * <pre>
+	 * l(t) = | x | = rho * |  cos(theta) | + t * | sin(theta) |
+	 *        | y |         | -sin(theta) |       | cos(theta) |
+	 * </pre>
+	 * <p>
+	 * In other words, {@code rho} represents the signed minimum distance from
+	 * the image origin to the line, and {@code theta} indicates the angle
+	 * between the row-axis and the minimum offset vector.
+	 * </p>
+	 * <p>
+	 * For a given point, then, votes are placed along the curve
+	 * </p>
+	 *
+	 * <pre>
+	 * rho = y * sin( theta ) + x * cos( theta )
+	 * </pre>
+	 * <p>
+	 * It is important to note that the interval of the first dimension of the
+	 * vote space image is NOT {@code [-maxRho, maxRho)} but instead
+	 * {@code [0, maxRho * 2)}; the same applies to the second dimension of the
+	 * vote space as well. Thus if {@link HoughTransforms#pickPeaks} is not used
+	 * to retrieve the maxima from the vote space, the vote space will have to
+	 * be translated by {@code -maxRho} in dimension 0 to get the correct
+	 * {@code rho} and {@code theta} values from the vote space.
+	 * </p>
+	 *
+	 * @param input
+	 *            - the {@link RandomAccessibleInterval} to run the Hough Line
+	 *            Transform over
+	 * @param votespace
+	 *            - the {@link RandomAccessibleInterval} in which the results
+	 *            are stored
+	 * @param nTheta
+	 *            - the number of bins for theta resolution
+	 * @param nRho
+	 *            - the number of bins for rho resolution
+	 * @param filter
+	 *            - a {@link Predicate} judging whether or not the a value is
+	 *            above the minimum value allowed by the populator. Any input
+	 *            less than or equal to this value will be disregarded by the
+	 *            populator.
+	 */
+	public static < T, U extends IntegerType< U > > void voteLines(
+			final RandomAccessibleInterval< T > input,
+			final RandomAccessibleInterval< U > votespace,
+			final int nTheta,
+			final int nRho,
+			final Predicate< T > filter )
+	{
+
+		final long[] dims = new long[ input.numDimensions() ];
+		input.dimensions( dims );
+
+		final double minRho = -computeLength( dims );
+		final double dRho = 2 * computeLength( dims ) / nRho;
+
+		final double minTheta = -Math.PI / 2;
+		final double dTheta = Math.PI / nTheta;
+
+		final double[] rho = new double[ nRho ];
```

Comment:
- `double[] rho` is never used, can be removed
 - `double[] theta` caches theta but the expensive caclulations are `Math.sin( theta )` and `Math.cos( theta )`. Cache these instead.